src/daemon/rpmostreed-os-experimental.cxx

```diff
@@ -165,11 +166,102 @@ osexperimental_handle_live_fs (RPMOSTreeOSExperimental *interface,
   return TRUE;
 }
 
+static gboolean
+osexperimental_handle_download_packages (RPMOSTreeOSExperimental *interface,
+                                         GDBusMethodInvocation *invocation,
+                                         GUnixFDList* fds,
+                                         const gchar * const *queries)
+{
+  g_autoptr(GCancellable) cancellable = g_cancellable_new ();
+  GError *local_error = NULL;
+  OstreeSysroot *sysroot = rpmostreed_sysroot_get_root (rpmostreed_sysroot_get ());
+  OstreeDeployment *booted_deployment = ostree_sysroot_get_booted_deployment (sysroot);
+  const char *osname = ostree_deployment_get_osname (booted_deployment);
+  GUnixFDList *out_fd_list = g_unix_fd_list_new ();
+  const gchar *path;
+  g_autoptr(GPtrArray) dnf_pkgs = g_ptr_array_new_with_free_func (g_object_unref);
+  DnfPackage *pkg = NULL;
```

Comment:
You should be able to move this variable declaration to the scopes where it's used.